maestro/config/exporter_builder.go

```diff
@@ -21,11 +21,25 @@ type PrometheusExporterConfig struct {
 }
 
 func (p *PrometheusExporterConfig) GetExportersFromMetadata(config types.Metadata, authenticationExtensionName string) (Exporters, string) {
-	endpointCfg := config.GetSubMetadata("exporter")["remote_host"].(string)
+	exporterSubMeta := config.GetSubMetadata("exporter")
+	if exporterSubMeta == nil {
+		return Exporters{}, ""
+	}
+	endpointCfg := exporterSubMeta["remote_host"].(string)
+	customHeaders := exporterSubMeta["headers"]
```

Comment:
nil pointer here if doesnt exist headers index in sink config map[string]interface{}

Reply:
that's why in the next line there is a check for customHeaders to be nil.